internal/agent/device/hook/actions.go

```diff
@@ -259,3 +367,53 @@ func (e *executableActionHook) OnChange(ctx context.Context, path string) error
 
 	return nil
 }
+
+func logAndReturn(log *log.PrefixLogger, format string, args ...any) error {
+	log.Errorf(format, args...)
+	return fmt.Errorf(format, args...)
+}
+
+func podmanComposeUp(ctx context.Context, path string, exec executer.Executer, log *log.PrefixLogger) error {
+	ctx, cancel := context.WithTimeout(ctx, time.Minute)
+	defer cancel()
+
+	_, stderr, exitCode := exec.ExecuteWithContext(ctx, "podman-compose", "-f", path, "up", "-d")
+	if exitCode != 0 {
+		return logAndReturn(log, "podman-compose up hook for path %s failed with exitCode %d: %s", path, exitCode, stderr)
+	}
+	return nil
+}
+
+func podmanComposeDown(ctx context.Context, path string, exec executer.Executer, log *log.PrefixLogger) error {
+	ctx, cancel := context.WithTimeout(ctx, time.Minute)
+	defer cancel()
+
+	_, stderr, exitCode := exec.ExecuteWithContext(ctx, "podman-compose", "-f", path, "down")
+	if exitCode == 0 {
+		return nil
+	}
+	_, err := os.Stat(path)
+	if !os.IsNotExist(err) {
+		return logAndReturn(log, "podman-compose down failed for existing path %s with exit code %d: %s", path, exitCode, stderr)
+	}
+	psStdout, psStderr, psExitCode := exec.ExecuteWithContext(ctx, "podman", "ps", "-a", "--format=json")
+	if psExitCode != 0 {
+		log.Errorf("podman ps --all failed: %s", psStderr)
+		return logAndReturn(log, "podman-compose down failed for path %s with exit code %d: %s", path, exitCode, stderr)
```

Comment:
instead of logAndReturn func can we return the error then log it?

Reply:
OK